### DISCUSSION TITLE
set_out_position and Wayland

### DISCUSSION THREAD

**Discussion by Vadoola** [rust-windowing/winit#2920] [Q&A] [answered]
I'm not very familiar with the internals of winit, and have never tried using it directly.

I've been playing around with Slint UI recently and trying to learn it.  One of the projects I've been doing is creating a frameless window and creating the "Title Bar" from within Slint so it can be themed with the program. This of course requires handling some things myself such as closing, minimizing and moving the window around.

My initial logic for handling moving the Window around has been working fine, however my initial testing was being done on Windows.

I discovered my logic for moving the window worked under Linux (on Wayland) if I used the Qt backend for Slint, but not the Winit backend. (Both Winit and Qt backend worked on X11)

I opened an [Issue ](https://github.com/slint-ui/slint/issues/3032) with Slint to try and get some more information although I was wondering if it was a winit issue.

It looks like the main reason it doesn't work is because [set_out_position](https://docs.rs/winit/latest/winit/window/struct.Window.html#method.set_outer_position) which is used by the Slint set_position function is unsupported on Wayland.

This is obviously a known issue as it is listed in the docs. What I'm trying to figure out is what the reason for this is? Has it simply not been implemented yet, is it a limitation of the Wayland protocol? I'm just trying to get a better idea of why this limitation, and what my possible options are for working around it.

**Comment by kchibisov** [comment, marked as answer]
>is it a limitation of the Wayland protocol

Yes, you don't have information where your window is or to set position.

Use https://docs.rs/winit/0.28.6/winit/window/struct.Window.html#method.drag_window and https://docs.rs/winit/0.28.6/winit/window/struct.Window.html#method.drag_resize_window

**Reply by Vadoola** [reply]
Thanks, it sounds like I'll have to put in some cfg guards depending on the slint backend or logic to specify that it can only use the winit backend, and then rework my move logic. Appreciate the help

**Reply by kchibisov** [reply]
`draw_window` works for every backend including windows.

**Reply by Vadoola** [reply]
good to know. I was already leaning towards changing my logic to just use drag_window.
